### PR TITLE
Add AppWorld

### PR DESCRIPTION
**Pull request by HarshTrivedi**
Thanks for setting up this repository! Please consider adding AppWorld to this list. I have inferred the format based on the surrounding papers, but feel free to change it as you prefer. Here are all the relevant links:

🔗 Website: https://appworld.dev/
📄 Paper: https://arxiv.org/abs/2407.18901
🐦 Tweet: https://x.com/harsh3vedi/status/1818311843976233198
💬 Blog: https://appworld.dev/blog
🎬 Video(s): https://appworld.dev/video
🌎 Code: https://github.com/stonybrooknlp/appworld
🧭 Data (task, trajectories) explorer, playground: https://appworld.dev/task-explorer
🔍 API explorer: https://appworld.dev/api-explorer
📊 Leaderboard: https://appworld.dev/leaderboard

**TLDR**: Introduces the AppWorld Engine, a high-fidelity execution environment of 9 day-to-day apps, operable via 457 APIs, populated with digital activities of 106 people living in a simulated world, and an associated benchmark of natural, diverse, and challenging autonomous agent tasks requiring rich and interactive coding. The tasks have a robust programmatic evaluation with state and execution-based unit tests.

### CODE REVIEW

**Comment by andreaskoepf** [comment]
@HarshTrivedi super cool project, love it! You put really a lot of effort into the env & eval!

**Comment by HarshTrivedi** [comment]
@andreaskoepf Yeah, it was indeed a lot of work. :-) Thank you very much for the kind words and for merging the PR!
